Code/GraphMol/SubstructLibrary/SubstructLibrary.h

```diff
@@ -678,6 +741,19 @@ class RDKIT_SUBSTRUCTLIBRARY_EXPORT SubstructLibrary {
     return rdcast<unsigned int>(molholder->size());
   }
 
+  //! does error checking
+  void setSearchOrder(const std::vector<unsigned int> &order) {
+    for (const auto idx : order) {
+      if (idx >= mols->size()) {
+        throw IndexErrorException(idx);
+      }
+    }
+    searchOrder = order;
+  }
+  const std::vector<unsigned int> &getSearchOrder() const {
+    return searchOrder;
+  }
+  std::vector<unsigned int> &getSearchOrder() { return searchOrder; }
```

Comment:
Is there a reason to return the non-const? I expect this might be for serialization purposes.

Reply:
> Is there a reason to return the non-const? I expect this might be for serialization purposes.

It's definitely required for serialization purposes, but it doesn't seem harmful to include this to allow easy edits of the search order when one has a non-const substruct library. Remember that the search order is equivalent to specifying a subset... easy subset edits can definitely be useful